src/GitHub.App/Authentication/TwoFactorRequiredUserError.cs

```diff
@@ -8,11 +8,18 @@ namespace GitHub.Authentication
     public class TwoFactorRequiredUserError : UserError
     {
         public TwoFactorRequiredUserError(TwoFactorAuthorizationException exception)
+            : this(exception, exception.TwoFactorType)
+        {
+        }
+
+        public TwoFactorRequiredUserError(
+            TwoFactorAuthorizationException exception,
+            TwoFactorType twoFactorType)
             : base(exception.Message, innerException: exception)
         {
             Guard.ArgumentNotNull(exception, nameof(exception));
 
-            TwoFactorType = exception.TwoFactorType;
+            TwoFactorType = twoFactorType;
```

Comment:
Added this extra constructor because the detection of the two-factor type is deep within Octokit's header parsing code. This allows us to easily override that in unit tests.